### ISSUE TITLE
Text property highlight doesn't override a sign highlight over a tab character

### ISSUE THREAD

**Issue by yegappan** [vim/vim#12584]
### Steps to reproduce

Source the following script:

```
func Test()
    split Xbuffer
    call setline(1, ['one', "\ttab", '        space', 'three'])
    call prop_type_add('Prop1', #{highlight: 'Search', override: v:true})
    sign define sign1 text=>> linehl=DiffAdd
    sign place 10 line=2 name=sign1
    sign place 20 line=3 name=sign1
    call prop_add(1, 1, #{end_lnum: 4, end_col: 5, type: 'Prop1'})
endfunc
call Test()
```

### Expected behaviour

The text property highlight group should override the sign line highlight group.  In the second line, before "tab", the sign highlight group is used.  But the first column uses the text property highlight group and the columns after the tab character use the text property highlight group.

### Version of Vim

9.0.1660

### Environment

Operating System: Ubuntu 22.04
Terminal: GNOME Terminal
Value of $TERM: xterm-256color

**Comment by chrisbra** [comment]
I am not sure, I don't know text properties well yet. I think the following patch fixes it:
```patch
diff --git a/src/drawline.c b/src/drawline.c
index 0638c5682..a28bc40a7 100644
--- a/src/drawline.c
+++ b/src/drawline.c
@@ -2023,7 +2023,13 @@ win_line(
                    ++text_prop_next;
                }

-               if (wlv.n_extra == 0 || !wlv.extra_for_textprop)
+               if (wlv.n_extra == 0 ||
+                       (!wlv.extra_for_textprop
+#ifdef FEAT_PROP_POPUP
+                        && !(text_prop_type != NULL &&
+                            text_prop_flags & PT_FLAG_OVERRIDE)
+#endif
+                   ))
                {
                    text_prop_attr = 0;
                    text_prop_attr_comb = 0;

```

I also noticed, that when use `:set list` the highlighting would also not be ovewritten by the text_property. Not sure if it should be or not

**Comment by yegappan** [comment]
Thanks @chrisbra for the patch.  I can confirm that this patch fixes the reported problem.

**Comment by chrisbra** [comment]
I will create a PR with this fix soon.

**Comment by chrisbra** [comment]
actually, I think there is at least one additional error. The Sign line highlighting is not respected. It should use the Search highlighting after last text character in the text and continue to use the DiffAdd highlighting and not the Search highlighting, right?

**Comment by chrisbra** [comment]
fixed by dbeadf05b6a152e7d9c5cc23d9202057f8e99884
